src/content/docs/tutorials/plot-data-with-charts.mdoc

```diff
@@ -0,0 +1,376 @@
+
+{% aside type="note" title="Bar Charts > Pie Charts" %}
+**Use bar charts when you can and pie charts when you must**. Why? Pie charts
+are often considered inferior to bar charts because they rely on human
+perception of angles, which is less accurate than judging lengths, making
+comparisons between categories harder. Bar charts allow for quick, precise
+comparisons, handle many categories cleanly, and often do not require labels,
+while pie charts become cluttered and confusing with numerous slices. Bar charts
+also allow easy sorting, are more space-efficient, and tend to stay cleaner
+without unnecessary visual distractions like 3D effects. Overall, bar charts
+communicate data more clearly, accurately, and efficiently than pie charts.
+{% /aside %}
+
+### Plot metrics as line chart
+
+Line charts come in handy when visualizing data trends over a continuous scale,
+such as time series data.
+
+{% steps %}
+
+1. **Shape your data**: For our line chart demo, we'll use some internal node
+   metrics provided by the [`metrics`](../reference/operators/metrics) operator.
+   Let's look at the RAM usage of the node:
+
+   ```tql
+   metrics "process"
+   drop swap_space, open_fds
+   head 3
+   ```
+
+   ```
+   {timestamp: 2025-04-27T18:16:17.692Z, current_memory_usage: 2363461632, peak_memory_usage: 4021136}
+   {timestamp: 2025-04-27T18:16:18.693Z, current_memory_usage: 2366595072, peak_memory_usage: 4021136}
+   {timestamp: 2025-04-27T18:16:19.694Z, current_memory_usage: 2385154048, peak_memory_usage: 4021136}
+   ```
+
+2. **Plot the data**: Add the
+   [`chart_line`](../../reference/operators/chart_line) operator to visualize
+   the time series. We are going to plot the memory usage within the last day:
+
+   ```tql
+   metrics "process"
+   where timestamp > now() - 1d
+   chart_line x=timestamp, y=current_memory_usage
+   ```
+
+   {% aside type="danger" title="TODO" %}
+   Insert chart here.
+   {% /aside %}
+
+2. **Aggregate to reduce the resolution**: Plotting metrics with a 1-second
+   granularity over the course of a full day can make a line chart very noisy.
+   In fact, we have a total of 86,400 samples in our plot. This can make a line
+   chart quickly illegible. Let's reduce the noise by aggregating the samples
+   into 15-min buckets:
+
+   ```tql
+   metrics "process"
+   where timestamp > now() - 1d
+   set timestamp = timestamp.round(15min)
+   summarize timestamp, mem=mean(current_memory_usage)
+   chart_line x=timestamp, y=mem
+   ```
+
+   {% aside type="danger" title="TODO" %}
+   Insert chart here.
+   {% /aside %}
+
+   This looks a lot smoother! You can go one step further and move the
+   aggregation into the chart operator call:
+
+   ```tql
+   metrics "process"
+   where timestamp > now() - 1d
+   chart_line x=timestamp, y=mean(current_memory_usage), resolution=15min
+   ```
+
+   Not only does this make the pipeline more succinct, it also takes care of
+   some subtle issues:
+
+   1. A
+   2. B
+   3. C
```

Comment:
@dominiklohmann could you help me articulate here why I want to use the inline aggregation via `resolution`?